sdk/storage/azure-storage-file-share/azure/storage/fileshare/_directory_client_helpers.py

```diff
@@ -40,21 +40,10 @@ def _format_url(scheme: str, hostname: str, share_name: str, dir_path: str, quer
     return f"{scheme}://{hostname}/{quote(share_name)}{directory_path}{query_str}"
```

Comment:
I think we briefly talked about this, any suggestions for the ideal way to approach share name converting to bytes? I believe this still exists for backwards compatibility sake :)

`azure/storage/fileshare/_directory_client_helpers.py:36: error: Incompatible types in assignment (expression has type "bytes", variable has type "str")  [assignment]`

Reply:
Perhaps changing the typehint of `share_name` to `Union[str, bytes]`, it is possible that this may spawn more MyPy errors in return 😄 but worth a shot?

Reply:
That fixed the error, `mypy` is happy 😁